gui/src/segment.ts

```diff
@@ -0,0 +1,135 @@
+
+            this._x = new ValueAndUnit(0);
+            this._y = new ValueAndUnit(0);
+
+            this._point = new Vector2(0, 0);
+        }
+
+        get x(): string | number {
+            return this._x.toString(this._multiLine._host);
+        }
+
+        set x(value: string | number) {
+            if (this._x.toString(this._multiLine._host) === value) {
+                return;
+            }
+
+            if (this._x.fromString(value)) {
+                this._multiLine._markAsDirty();
+            }
+        }
+
+        get y(): string | number {
+            return this._y.toString(this._multiLine._host);
+        }
+
+        set y(value: string | number) {
+            if (this._y.toString(this._multiLine._host) === value) {
+                return;
+            }
+
+            if (this._y.fromString(value)) {
+                this._multiLine._markAsDirty();
+            }
+        }
+
+        get control(): Nullable<Control> {
+            return this._control;
+        }
+
+        set control(value: Nullable<Control>) {
+            if (this._control === value) {
+                return;
+            }
+
+            if (this._control && this._controlObserver) {
+                this._control.onDirtyObservable.remove(this._controlObserver);
+
+                this._controlObserver = null;
+            }
+
+            this._control = value;
+
+            if (this._control) {
+                this._controlObserver = this._control.onDirtyObservable.add(this._multiLine.onSegmentUpdate);
+            }
+
+            this._multiLine._markAsDirty();
+        }
+
+        get mesh(): Nullable<AbstractMesh> {
+            return this._mesh;
+        }
+
+        set mesh(value: Nullable<AbstractMesh>) {
+            if (this._mesh === value) {
+                return;
+            }
+
+            if (this._mesh && this._meshObserver) {
+                this._mesh.getScene().onAfterCameraRenderObservable.remove(this._meshObserver);
+            }
+
+            this._mesh = value;
+
+            if (this._mesh) {
+                this._meshObserver = this._mesh.getScene().onAfterCameraRenderObservable.add(this._multiLine.onSegmentUpdate);
+            }
+
+            this._multiLine._markAsDirty();
+        }
+
+        translate(): Vector2 {
+            this._point = this._translatePoint();
+
+            return this._point;
+        }
+
+        private _translatePoint(): Vector2 {
+            if (this._mesh != null) {
+                return this._multiLine._host.getProjectedPosition(this._mesh.getBoundingInfo().boundingSphere.center, this._mesh.getWorldMatrix());
+            }
+            else if (this._control != null) {
+                return new Vector2(this._control.centerX, this._control.centerY);
+            }
+            else {
+                //TODO: extract number from px and % - eventually in order to support %
+                return new Vector2(this.x as number, this.y as number); //temp - as number

```

Comment:
I'd like to translate Segment x and y (ValueAndUnit) to pixels. Do I need to manually pass the adt size to ValueAndUnit/getValueInPixel()? Isn't there a better way?

For instance, in a 1000x1000 canvas size, I'd like to translate { x: '50%', y: '50px' } to { x: 500, y: 50 }.

Reply:
By the way I changed my mind regarding naming :) I prefer AnchorPoint instead of segment as it is not really a segment :)
yes for your question: you have to give adtSize to getValueInPixel()

Reply:
Hmm Anchor isn't quite correct either in my opinion, let me think about it too :)

Reply:
How about simply MultiLinePoint? Or PointInfo / PointData / DynamicPoint / AdvancedPoint?

Reply:
MultiLinePoint is great